### DISCUSSION TITLE
Branch

### DISCUSSION THREAD

**Discussion by KyungsooJ** [community/community#21709] [Other Feature Feedback, Questions, & Ideas]
<p><strong>Hi everyone,</strong></p>
<p><strong>I have some question about the branch</strong></p>
<p><strong>1. I searched about the Branch in Google. There are so many explained about this function, but I don’t know what’s the relation with Merge. Every explains were written with merge.</strong></p>
<p><strong>2. How can i know at git that the branch has already been merged as a master?</strong></p>
<p><strong>3. What is the main reason to use brunch in git?</strong></p>
<p>Thank you</p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a href="/t5/user/viewprofilepage/user-id/37993">@joe97228</a>,</p>
<p>Welcome to the Github Community Forum!</p>
<blockquote>
<p><a class="mention" href="/kyungsooj">@kyungsooj</a> wrote:</p>
</blockquote>
<blockquote>
<p><strong>1. I searched about the Branch in Google. There are so many explained about this function, but I don’t know what’s the relation with Merge. Every explains were written with merge.</strong></p>
</blockquote>
<p>_ <strong>The branch without merge is like having an hair comb with no hair.</strong> _ Yes, you can use it but you never appreciate it. With example maybe you can understand better. Think about a git repository with some commits:</p>
<p>              master -&gt;  A—B---C—D</p>
<p>The A, B, C and D are commits that perfectly solve a single problem and you can present as a finished project as is. Now you want to try to create a <strong>new feature</strong> without touch the main project. <strong>Without git</strong> you can copy and paste the project in another directory and start to make some changes, but <strong>we are brave enough to try using git(?)</strong>. With git you need to create a new history path with branches, in our example we are going to create a branch called new.feature:</p>
<p>                            new.feature -&gt;  E—F---G</p>
<p>                                  git branch  /<br></p>
<p>              master -&gt;  A—B---C—D</p>
<p>The new commits E, F and G solved the new feature we want to create and we are ready to present this new version of the project. But how can we manage to substitute the last commit D of the master branch? With <strong>the merge command</strong> :</p>
<p>                            new.feature -&gt;  E—F---G</p>
<p>                                  git branch  /             \   git merge<br></p>
<p>              master -&gt;  A—B---C—D---<br></p>
<p>The new history is this one with the E, F and G commits head after A, B, C and D commits:</p>
<p>              master -&gt;  A—B---C—D---E—F---G</p>
<p>Hope you understand.<br></p>
<p>PS: Ask if you want step to step git command tutorial for this example.</p>
<blockquote>
<p><a class="mention" href="/kyungsooj">@kyungsooj</a> wrote:</p>
</blockquote>
<blockquote>
<p><strong>2. How can i know at git that the branch has already been merged as a master?</strong></p>
</blockquote>
<p>If you are using the git in console/terminal you can simply use the log command of the branch and <strong>compare with the log</strong> of the branch master. If commits of the branch are the same in the master branch so that branch is already merge. Alternatively you can <strong>try to merge</strong> that branch into master without create commit using the command merge with argument <strong>–squash</strong> and watch if there are some changes to stage.</p>
<p>If the question is: How can i know at <strong>GITHUB</strong> that the branch has already been merged as master? The answer is you can see the branch status in the overview into the web ui at <a target="_blank">https://github.com/&lt;user_name&gt;/&lt;repository_name&gt;/branches</a>.</p>
<blockquote>
<p><a class="mention" href="/kyungsooj">@kyungsooj</a> wrote:</p>
</blockquote>
<blockquote>
<p><strong>3. What is the main reason to use brunch in git?</strong></p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<p>Because the branch is the <strong>Thor’s hammer</strong> of git. To be more precise branching is used to work in team, solve issues without touch the master version, try to add new features and discard them if are not required and so on. There is nothing to stop you from using only the master branch, <strong>even Thor can beat small enemies without the hammer</strong>.</p>
<p>Ask if you have more questions…</p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>
